graffer-ql/README.md

```diff
@@ -0,0 +1,130 @@
+  Copyright 2016 Crown Copyright
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+GrafferQL
+==================================
+
+GrafferQL is a library that configures the GraphQL-java library using a Gaffer Schema.
+
+This allows you to write GraphQL queries which will be converted to Gaffer operations.
+
+The Gaffer Element properties returned are used to generate GraphQL compliant responses.
+
+Currently the library simply configures an instance graphql.GraphQL with a gaffer.store.schema.Schema.
+There is no REST interface for this yet so native Java access is the only way to use it.
+
+Setup
+------------------
+An example can be seen in graphql.gaffer.GrafferQlTest.
+
+1. Create a Gaffer graph in the normal way
+
+    Graph graph = new Graph.Builder()
+        .storeProperties(...stuff...)
+        .addSchemas(...stuff...)
+        .build();
+        
+2. Use the GafferQLSchemaBuilder class to build a graphql.GraphQL instance
+
+    GraphQL graphQL = new GafferQLSchemaBuilder()
+        .gafferSchema(graph.getSchema())
+        .build();
+        
+For each request/operation chain you wish to run, do the following
+1. Construct a User to act as
+
+    User user = new User.Builder()
+        .userId("me")
+        .dataAuth("see_everything")
+                
+2. Build a context for GrafferQL to use, this requires
+  a. The gaffer.graph.Gaffer graph
+  b. The gaffer.user.User
+
+    GrafferQLContext context = new GrafferQLContext.Builder()
+        .graph(graph) // will be same value for all requests to this graph
```

Comment:
yeah agreed, I struggled to get it to play ball. Sometimes it puts text in the little grey boxes, but not everywhere. There seems to be no way to explicitly mark it up as 'code'